konduit-serving-pipeline/src/main/java/ai/konduit/serving/pipeline/impl/data/helpers/ProtobufUtils.java

```diff
@@ -18,22 +18,175 @@
 package ai.konduit.serving.pipeline.impl.data.helpers;
 
 import ai.konduit.serving.pipeline.api.data.*;
-import ai.konduit.serving.pipeline.api.format.ImageConverter;
 import ai.konduit.serving.pipeline.impl.data.JData;
 import ai.konduit.serving.pipeline.impl.data.Value;
 
 import java.nio.ByteBuffer;
 import java.util.*;
 
+import ai.konduit.serving.pipeline.impl.data.box.BBoxCHW;
+import ai.konduit.serving.pipeline.impl.data.box.BBoxXY;
+import ai.konduit.serving.pipeline.impl.data.image.Png;
 import ai.konduit.serving.pipeline.impl.data.ndarray.SerializedNDArray;
 import ai.konduit.serving.pipeline.impl.data.protobuf.DataProtoMessage;
-import ai.konduit.serving.pipeline.impl.format.JavaImageConverters;
+import ai.konduit.serving.pipeline.impl.data.wrappers.ListValue;
 import com.google.protobuf.ByteString;
+import lombok.val;
 import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
 
 public class ProtobufUtils {
 
-    public static DataProtoMessage.DataMap convertJavaToProtobufData(Map<String,Value> dataMap) {
+    private static String PNG = "PNG";
+
+    private static List<ByteString> ndArrayToByteStringList(SerializedNDArray sn) {
+        byte[] bufferBytes = new byte[sn.getBuffer().remaining()];
```

Comment:
Let's add a `sn.getBuffer().rewind();` before we use it here to make sure we get all values.

Reply:
Fixed.